frontend/src/components/dashboard/ProgressList.tsx

```diff
@@ -181,17 +181,22 @@ const ProgressList = ({
     isLoading: isLoadingSolutions,
   } = useAllSessionSolutions(classId, sessionId);
 
-  const progress = useMemo(() => {
-    if (!klass || !session || !solutions) {
-      return [];
-    }
+  const studentList = useMemo(() => {
+    if (!klass || !solutions) return new Set<number>();
 
     const studentIds = new Set([
       ...klass.students.map((student) => student.studentId),
       ...solutions.flatMap((s) => s.solutions.map((s) => s.studentId)),
     ]);
+    return studentIds;
```

Comment:
I think the rest will be easier to read if we already convert it to an array here:

```suggestion
    // convert the set to an array
    return [...studentIds];
```

(you will need to change the early return too)